src/OrchardCore.Modules/OrchardCore.Tenants/Workflows/Activities/CreateTenantTask.cs

```diff
@@ -0,0 +1,114 @@
+        public override LocalizedString Category => T["Tenant"];
+        public override LocalizedString DisplayText => T["Create Tenant Task"];
+
+        public string ContentType
+        {
+            get => GetProperty<string>();
+            set => SetProperty(value);
+        }
+        
+        public WorkflowExpression<string> RequestUrlPrefix
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> RequestUrlHost
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> DatabaseProvider
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> ConnectionString
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> TablePrefix
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> RecipeName
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+        
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(T["Done"]);
+        }
+
+        public async override Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var tenantNameTask = _expressionEvaluator.EvaluateAsync(TenantName, workflowContext);
+            var requestUrlPrefixTask = _expressionEvaluator.EvaluateAsync(RequestUrlPrefix, workflowContext);
+            var requestUrlHostTask = _expressionEvaluator.EvaluateAsync(RequestUrlHost, workflowContext);
+            var databaseProviderTask = _expressionEvaluator.EvaluateAsync(DatabaseProvider, workflowContext);
+            var connectionStringTask = _expressionEvaluator.EvaluateAsync(ConnectionString, workflowContext);
+            var tablePrefixTask = _expressionEvaluator.EvaluateAsync(TablePrefix, workflowContext);
+            var recipeNameTask = _expressionEvaluator.EvaluateAsync(RecipeName, workflowContext);
+
+            await Task.WhenAll(tenantNameTask, requestUrlPrefixTask, requestUrlHostTask, databaseProviderTask, connectionStringTask, tablePrefixTask, recipeNameTask);
+
+            var shellSettings = new ShellSettings();
+
+            if (!string.IsNullOrWhiteSpace(tenantNameTask.Result))
+            {
+                shellSettings = new ShellSettings
+                {
```

Comment:
Need to be reviewed, here maybe we need

                // Creates a default shell settings based on the configuration.
                var shellSettings = _shellSettingsManager.CreateDefaultSettings();

If no task name you still save the settings